### PR TITLE
Revert ml stuff

### PR DESCRIPTION
**Pull request by jannes-m**

### CODE REVIEW

**Comment by jannes-m** [comment]
In the latest mlr3 version is a breaking change (https://github.com/mlr-org/mlr3/releases/tag/v0.21.0)  how fallback learners are defined (see also #1111). I have addressed this, let's what comes next -:)

**Review by Robinlovelace — APPROVED** [review]

File: 15-eco.Rmd [hunk 1]
```diff
@@ -515,25 +515,20 @@ autotuner_rf$train(task)
...
-#> 1:     4           0.878             7          <list[4]> <list[3]>     0.368
 ```
 
 ### Predictive mapping
 
 The tuned hyperparameters\index{hyperparameter} can now be used for the prediction.
 To do so, we only need to run the `predict` method of our fitted `AutoTuner` object.
 
-```{r 15-eco-27, cache=TRUE, cache.lazy=FALSE, warning=FALSE, eval=FALSE}
+```{r 15-eco-27, cache=TRUE, cache.lazy=FALSE, warning=FALSE}
```
Comment (Robinlovelace) [inline 15-eco.Rmd:531]: Actions still having issues here:

```
62/70 [15-eco-27]        
Error in `encapsulate()`:
! Assertion on 'method' failed: Must be a subset of {'none','try','evaluate','callr'}, not 'NULL'.
Backtrace:
  1. base::eval(...)
  2. base::eval(...)
  4. rmarkdown (local) `<fn>`(...)
  5. knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
  6. knitr:::process_file(text, output)
     ...
 17. base::withRestarts(...)
 18. base (local) withRestartList(expr, restarts)
 19. base (local) withOneRestart(withRestartList(expr, restarts[-nr]), restarts[[nr]])
 20. base (local) docall(restart$handler, restartArgs)
 22. evaluate (local) fun(base::quote(`<smplErrr>`))
Warning messages:
1: In warn_deprecated("Task$data_formats") :
  Task$data_formats is deprecated and will be removed in the future.
2: In warn_deprecated("DataBackend$data_formats") :
  DataBackend$data_formats is deprecated and will be removed in the future.
3: In warn_deprecated("Learner$data_formats") :
  Learner$data_formats is deprecated and will be removed in the future.

Quitting from lines 532-534 [15-eco-27] (15-eco.Rmd)
Execution halted
Error in Rscript_render(f, render_args, render_meta, add1, add2) : 
  Failed to compile 15-eco.Rmd
Calls: <Anonymous> -> render_new_session -> Rscript_render
Execution halted
```

Source: https://github.com/geocompx/geocompr/actions/runs/11033004044/job/30643176006?pr=1124#step:4:2039

We should fix those, either in the code here or in the code that defines the Docker images or workflows before merging.

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Let's merge after actions passing.

File: 15-eco.Rmd (hunk 1 above)
Comment (jannes-m) [reply 15-eco.Rmd:531]: seems to be fixed now

**Comment by Robinlovelace** [comment]
Great job, thanks Jannes, will merge!

**Pull request merged**